packages/amplify-graphql-auth-transformer/src/vtl-generator/rds/resolvers/common.ts

```diff
@@ -95,7 +95,7 @@ const convertAuthRoleToVtl = (
           type: str(role.strategy),
           provider: str('iam'),
           roleArn: role.strategy === 'public' ? ref('ctx.stash.unauthRole') : ref('ctx.stash.authRole'),
-          cognitoIdentityPoolId: hasIdentityPoolId ? ref('ctx.stash.identityPoolId') : nul(),
+          ...(role.strategy === 'private' && { cognitoIdentityPoolId: identityPoolId ? str(identityPoolId) : nul() }),
```

Comment:
You need to reference the identityPoolId in stash, right? I don't think that value `identityPoolId` you're referencing exists. This should fail to build if I'm not mistaken.

Reply:
Yeah, just double-checked, and it looks like the build is failing on this.

Reply:
Ah, looks rebase didn't go well. Looking into it.

Reply:
PR has been updated now with the correct changes.